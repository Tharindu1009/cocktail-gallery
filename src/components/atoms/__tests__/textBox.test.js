import { render, cleanup, fireEvent } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import TextBox from "../textBox/TextBox";

describe('TextBox', () => {
    afterEach(() => {
        cleanup();
    })
    
    const mockHandleChange = jest.fn();
    
    it('should render textBox component correctly', () => {
        const { getByTestId } = render(<TextBox id="search" value="ZENMEISTER" size="small"
            placeholder="Search for a cocktail" onChange={mockHandleChange} />);
        const textBoxElement = getByTestId('textBox');
        expect(textBoxElement).toBeInTheDocument();
    });

    it('updates the textBox value', () => {
        const { getByRole } = render(<TextBox id="search" value="ZENMEISTER" size="small"
            placeholder="Search for a cocktail" onChange={mockHandleChange} />);
        const input = getByRole('textbox');
        fireEvent.change(input, { target: { value: 'ZENMEISTER' } });
        expect(input.value).toBe('ZENMEISTER');
    });

    it('calls the handleChange function three times when the input is changed', () => {
        const { getByRole } = render(<TextBox id="search" value="ZENMEISTER" size="small"
            placeholder="Search for a cocktail" onChange={mockHandleChange} />);
        const input = getByRole('textbox');
        fireEvent.change(input, { target: { value: 'BUCCANEER' } });
        fireEvent.change(input, { target: { value: 'JAMAICAN COFFEE' } });
        expect(mockHandleChange).toHaveBeenCalledTimes(2);
    });

    it('matches textBox snapshot', () => {
        const tree = TestRenderer.create(<TextBox id="search" value="ZENMEISTER" size="small"
            placeholder="Search for a cocktail" onChange={mockHandleChange} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
