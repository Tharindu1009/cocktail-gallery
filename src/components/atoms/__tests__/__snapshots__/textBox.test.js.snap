// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox matches textBox snapshot 1`] = `
<div
  className="textBox"
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root textBox css-wb57ya-MuiFormControl-root-MuiTextField-root"
    data-testid="textBox"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="off"
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="search"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Search for a cocktail"
        required={false}
        type="text"
        value="ZENMEISTER"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`matches textBox snapshot 1`] = `
<div
  className="textBox"
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root textBox css-wb57ya-MuiFormControl-root-MuiTextField-root"
    data-testid="textBox"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="off"
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="username"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Username"
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
